File: runner/js/propose.mjs
#!/usr/bin/env node
/** Emit {vector id: hash-of-our-answer} for everything this side can compute. ⚠ Silent on the rest. */
import { readFileSync, readdirSync } from 'node:fs'
import { fileURLToPath } from 'node:url'
import { dirname, join } from 'node:path'
import { canonHash, OPS } from './core.mjs'
const ROOT = join(dirname(fileURLToPath(import.meta.url)), '..', '..')
const out = {}
for (const f of readdirSync(join(ROOT, 'vectors')).filter(x => x.endsWith('.json')).sort()) {
  for (const v of JSON.parse(readFileSync(join(ROOT, 'vectors', f), 'utf8')).vectors) {
    const fn = OPS[v.op]; if (!fn) continue
    try { out[v.id] = canonHash(fn(v.in)) }
    catch (e) { console.error(`# ${v.id} threw ${e.message}`) }
  }
}
console.log(JSON.stringify(out))

File: runner/python/propose.py
#!/usr/bin/env python3
"""Emit {vector id: hash-of-our-answer} for everything this side can compute. ⚠ Silent on the rest."""
import json, pathlib, sys
sys.path.insert(0, str(pathlib.Path(__file__).parent))
from run import dispatch, canon_hash, NotImplementedOp
ROOT = pathlib.Path(__file__).resolve().parents[2]
out = {}
for f in sorted((ROOT / 'vectors').glob('*.json')):
    for v in json.loads(f.read_text())['vectors']:
        try: out[v['id']] = canon_hash(dispatch(v['op'], v['in']))
        except NotImplementedOp: pass
        except Exception as e: print(f"# {v['id']} threw {e}", file=sys.stderr)
print(json.dumps(out))
